cpp/src/common/tsfile_common.h

```diff
@@ -287,15 +294,17 @@ struct ChunkMeta {
 };
 
 struct ChunkGroupMeta {
-    // std::string device_name_;
-    common::String device_name_;
+    std::shared_ptr<IDeviceID> device_name_;
+    common::String device_name_str_;
     common::SimpleList<ChunkMeta *> chunk_meta_list_;
 
     explicit ChunkGroupMeta(common::PageArena *pa_ptr)
-        : device_name_(), chunk_meta_list_(pa_ptr) {}
+        : chunk_meta_list_(pa_ptr) {}
 
-    FORCE_INLINE int init(const std::string &dev_name, common::PageArena &pa) {
-        return device_name_.dup_from(dev_name, pa);
+    FORCE_INLINE int init(std::shared_ptr<IDeviceID> device_id,
+                          common::PageArena &pa) {
+        device_name_ = device_id;
+        return device_name_str_.dup_from(device_id->get_device_name(), pa);
```

Comment:
Avoid using the string form. Concatenating the segments with "." may cause trouble if a segment contains "." already.